clients/roscpp/src/libros/callback_queue.cpp

```diff
@@ -378,8 +384,11 @@ CallbackQueue::CallOneResult CallbackQueue::callOneCB(TLS* tls)
       }
       else
       {
-        tls->cb_it = tls->callbacks.erase(tls->cb_it);
```

Comment:
I would think we should keep this erase in here... at least in the case where we can get a lock on the weak ptr.